Update readme (#138)
seamapi/ruby-next: README.md

## Description

[Seam](https://seam.co) makes it easy to integrate IoT devices with your applications.
This is an official SDK for the Seam API.
Please refer to the official [Seam Docs](https://docs.seam.co/latest/) to get started.

Parts of this SDK are generated from always up-to-date type information

> provided by [@seamapi/types](https://github.com/seamapi/types/) node package.

Reviewer: ```suggestion
provided by [@seamapi/types](https://github.com/seamapi/types/).
```

This ensures all API methods, request shapes, and response shapes are
accurate and fully typed.

## Installation

> Add this as a dependency to your project using [Bundler](https://bundler.io/) with:

Reviewer: ```suggestion
Add this as a dependency to your project using [Bundler] with:
```

Author: 5029d950489e7d896990f7c06788e351adb7c0bd

> ```bash

Reviewer: Minor point, but now this is the only repo that does this. I think I may have omitted it before because this is not actually bash due to the prompt `$`
```suggestion
```
```

Author: 10d27a358ea01bf7c1d3d2524c6e28d259a68234

$ bundle add seam
```

> ## Usage

Reviewer: ```suggestion
[Bundler]: https://bundler.io/

## Usage
```

Author: 5029d950489e7d896990f7c06788e351adb7c0bd

### Examples

> **Note:** _These examples assume `SEAM_API_KEY` is set in your environment._

Reviewer: Let's upgrade this and the other README to use the GH note syntax
![Uploading image.png…]()

```suggestion
> [!NOTE]
> These examples assume `SEAM_API_KEY` is set in your environment.
```

Author: b902492cc500c0d0e6d79b341a81b1b439c978ec

locks = seam.locks.list

if locks.empty?
  raise "No locks in this workspace"
end

lock = locks.first

begin
  seam.locks.unlock_door(
    device_id: lock.device_id,
    wait_for_action_attempt: {
      timeout: 5.0,
      polling_interval: 1.0
    }
  )

  puts "Door unlocked"
rescue Seam::ActionAttemptFailedError
  puts "Could not unlock the door"
rescue Seam::ActionAttemptTimeoutError
  puts "Door took too long to unlock"
end
```

### Interacting with Multiple Workspaces

Some Seam API endpoints interact with multiple workspaces. The `Seam::Http::SeamMultiWorkspace` client is not bound to a specific workspace and may use those endpoints with a personal access token authentication method.

A Personal Access Token is scoped to a Seam Console user. Obtain one from the Seam Console.

```ruby
# Pass as an option to the constructor
seam = Seam::Http::SeamMultiWorkspace.new(personal_access_token: "your-personal-access-token")

# Use the factory method
seam = Seam::Http::SeamMultiWorkspace.from_personal_access_token("your-personal-access-token")

# List workspaces authorized for this Personal Access Token
workspaces = seam.workspaces.list
```

### Webhooks

> The Seam API implements webhooks using [Svix](https://www.svix.com). This SDK exports a thin wrapper `Seam::Webhook` around the svix package. Use it to parse and validate Seam webhook events.

Reviewer: ```suggestion
The Seam API implements webhooks using [Svix](https://www.svix.com).
This SDK exports a thin wrapper `Seam::Webhook` around the svix package.
Use it to parse and validate [Seam webhook events](https://docs.seam.co/latest/developer-tools/webhooks)
```

Author: c7bf1a319dfa8291d4c6abe01aa5968aa714970e

> Refer to the [Svix docs on Consuming Webhooks](https://docs.svix.com/receiving/introduction) for an in-depth guide on best-practices for handling webhooks in your application.

Reviewer: ```suggestion
> [!TIP]
> This example is for [Sinatra](https://sinatrarb.com/), see the [Svix docs for more examples in specific frameworks](https://docs.svix.com/receiving/verifying-payloads/how).

```

Author: 44b0e437d538920f78f7a66dd0beb954da98e4a0

> Refer to the [Svix docs on Consuming Webhooks](https://docs.svix.com/receiving/introduction) for an in-depth guide on best-practices for handling webhooks in your application.

Reviewer: ```suggestion
Refer to the [Svix docs on Consuming Webhooks](https://docs.svix.com/receiving/introduction)
for an in-depth guide on best-practices for handling webhooks in your application.
```

Author: c7bf1a319dfa8291d4c6abe01aa5968aa714970e

> ```ruby

Reviewer: Just want to ask if you were able to run this locally :)

Author: Yep yep, I did test it locally. Though I only tested the error case which gets successfully caught by `Seam::WebhookVerificationError`. What's the way to pass correct headers for webhook verification to succeed?

Reviewer: Try this :) https://www.svix.com/guides/testing-webhooks-locally-svix-cli/

Author: Hmm, for some reason I'm getting "No matching signature found"

![image](https://github.com/user-attachments/assets/9868168d-d3ec-49b1-9deb-ec2225d87ff2)

Author: I finally made it work! Here are the fixes: 3ace78cfa9cb242f4a70c0ac1e280d3a9c25711f
